### DISCUSSION TITLE
Auto-populate fields as per URL parameters

### DISCUSSION THREAD

**Discussion by yatharth-zeotap** [Budibase/budibase#4156] [Help]
Hi everyone

I am stuck with navigating between pages and auto-populating the fields according to the parameters' values passed in the URL

Consider I have a screen like this.
![image](https://user-images.githubusercontent.com/74049974/150635344-fac40735-2c96-4f83-af97-31e5743d5c59.png)
And the 2nd screen to be this

![image](https://user-images.githubusercontent.com/74049974/150635354-1b47980e-936b-454a-922f-00755ccc0ce1.png)

Now on using the button I want to open a modal and populate the value from `COLUMN1` to the text field and `COLUMN2` to the paragraph field.
So in this case _what should be the URL_ that I should provide in the `Navigate To` action.

This is my UI structure.
![image](https://user-images.githubusercontent.com/74049974/150635374-d3dcbf07-af56-46a3-9aed-9980c5e8a385.png) ![image](https://user-images.githubusercontent.com/74049974/150635379-6b2881ac-425b-4009-ab8d-0183acbeb814.png)

Any help will be great.

**Comment by mjashanks** [comment]
@yatharth-zeotap so you would like to display an "Edit" screen, in a modal?

So, the screen that is to appear in the modal should have a URL like `/your-record/:id` the `:id` is a URL parameter, which we will use to load the record.

Your edit screen should have a structure a bit like this:

![image](https://user-images.githubusercontent.com/3524181/150768329-b79f8e82-14e7-4f22-ae74-df11658a56bb.png)

I.e. 
```
> Data Provider
  > Repeater 
     > Form
            ....fields
```

Your data provider should load the record from your table, and should have a filter like this:

![image](https://user-images.githubusercontent.com/3524181/150768505-4dd5f0a6-d5bf-42ad-96eb-124c241da207.png)

I.e. it's loading the record based on the `:id` passed in via the URL parameter.

That should be enough to get you going! Your Save button will need to include a `Close Screen Modal` action, to close the modal.

**Comment by yatharth-zeotap** [comment]
Hi @mjashanks, thanks for the response.

I figured out there I can set global variables and use them on a different page, which would solve the issue for me.

But I am facing another issue in doing that.
**_I am able to set up the global variable for a Postgres based data-provider but in the case of a REST data source that doesn't seem to work for me._**

These are the actions I have used.
![image](https://user-images.githubusercontent.com/74049974/151501720-f701a8e8-5c43-479f-b013-1def198d9299.png)

And this is my table
![image](https://user-images.githubusercontent.com/74049974/151501797-756a08cf-2102-4304-9c1d-dbbd216724c3.png)

**Reply by mjashanks** [reply]
Not 100% if this is your issue, but I think we have just fixed this - in Budibase Cloud & in `v1.0.46`. https://github.com/Budibase/budibase/issues/4243

You should be able to bind state to your REST API bindings, using the cog icon, in a dataprovider:

![image](https://user-images.githubusercontent.com/3524181/151598024-02fe3488-b186-44ee-844e-be49b017f5f9.png)

**Reply by yatharth-zeotap** [reply]
Thanks @mjashanks , that solved my issue

Can you help me with these also?
1. Can I make the table column values are now collapsable, If the values get large in size, they get represented in `....`
2. Can we add a search bar to the table?
3. For tables created using a REST data source, is there a way to paginate them? I am ending up with about 300 rows.
4. How to make all the pages navigatable, showing their navigatble button on top of the screen
Like in this image
![image](https://user-images.githubusercontent.com/74049974/151916671-70d54c9b-d6ac-4317-b09f-743dead6f37c.png)

**Reply by mjashanks** [reply]
> 1. Can I make the table column values are now collapsable, If the values get large in size, they get represented in ....

Not at the moment, we are planning some work to the table component in the next few months.

> 2. Can we add a search bar to the table?

https://docs.budibase.com/guides/add-a-dropdown-filter-to-a-table describes how you would do this. You can also use a `Filter` component as a more dynamic option.

> 3. For tables created using a REST data source, is there a way to paginate them? I am ending up with about 300 rows.

Our REST connection supports pagination. This was only released a few weeks ago, so you might need to update if running self-hosted:
![image](https://user-images.githubusercontent.com/3524181/151961664-e76e14e4-0eb7-4c56-b053-922294693365.png)

> 4. How to make all the pages navigatable, showing their navigatble button on top of the screen

See layouts: https://docs.budibase.com/building-apps/design/layouts

**Reply by yatharth-zeotap** [reply]
@mjashanks 


> > 2. Can we add a search bar to the table?
> 
> https://docs.budibase.com/guides/add-a-dropdown-filter-to-a-table describes how you would do this. You can also use a `Filter` component as a more dynamic option.
>
>>
I checked the documentation for the `Filter` component, but how do I use the `Options` while fetching the data from a REST source.
I need a dynamic filter such that I can see the whole table data as well as search for a particular value.


> 
> > 3. For tables created using a REST data source, is there a way to paginate them? I am ending up with about 300 rows.
> 
> Our REST connection supports pagination. This was only released a few weeks ago, so you might need to update if running self-hosted: ![image](https://user-images.githubusercontent.com/3524181/151961664-e76e14e4-0eb7-4c56-b053-922294693365.png)
>
>>
I couldn't find the documentation for the same, can you please help me with the process. What are these pagination types and how do I configure them


_**Also, can I modify font and size of table values ?**_

**Reply by yatharth-zeotap** [reply]
@mjashanks Can you please help with the above

Also, how do I make the table row value non-collapsible on hover? What should be the CSS for that ?

**Reply by mjashanks** [reply]
> how do I use the Options while fetching the data from a REST source.

I just tried this. I think I know what you mean...

The Dynamic Filter component only applies to the "Define Filters" filter (see below). You cannot use it to pass binding to your REST query (i.e. the cog icon)

![image](https://user-images.githubusercontent.com/3524181/156607995-b6da3ff0-1a0d-4446-af9c-19d6a3e243f7.png)

To pass values to bindings, you will have to setup the filter fields manually.

> What are these pagination types and how do I configure them

How does your REST API support pagination - then I can help you with trying to get Budibase to work with that :)

> Also, can I modify font and size of table values ?

Unfortunately not at the moment. We are just about to release a lot of table improvements, but that is not one of them - sorry: 
https://github.com/Budibase/budibase/pull/4608

> how do I make the table row value non-collapsible on hover?

I'm sorry, I don't know what you mean. Does a table row collapse on hover at the moment?

**Comment by yatharth-zeotap** [comment]
1.
> I'm sorry, I don't know what you mean. Does a table row collapse on hover at the moment?

<img width="202" alt="image" src="https://user-images.githubusercontent.com/74049974/156702109-c579e2ec-ac4e-4317-9c4e-59da30b80c00.png">

Can we hover over this column and see the full value of this field (It has been replaced with **...** )? What will be the custom CSS for this?



2. 
> How does your REST API support pagination - then I can help you with trying to get Budibase to work with that :)

My API doesn't support pagination, I wanted budibase to take the complete JSON body and show them on multiple table pages.

**Reply by mjashanks** [reply]
1.

Got it, thanks - in the next release you'll be able to increase the width of a column and apply text wrapping. I don't think there is a way to hover over to show it - though that's a good idea that I will suggest :)

2.

Ok - so, assuming that your API returns an array, then there is no need to use pagination inside the REST API setup. You can use the pagination inside the design section. On a Data Provider, you will see "Limit" field. This defines your page-size. This means that Budibase will pull back the entire contents of a request, and paginate it client-side. This assumes that your API is returning an array. If it's  not returning an array, you can use a transformer (inside the REST API query setup) to make it into an array.

![image](https://user-images.githubusercontent.com/3524181/156738451-46984a7a-9bfc-476d-b21a-95a9635694d3.png)

**Comment by yatharth-zeotap** [comment]
@mjashanks 

> Ok - so, assuming that your API returns an array, then there is no need to use pagination inside the REST API setup. You can use the pagination inside the design section. On a Data Provider, you will see "Limit" field. This defines your page-size. This means that Budibase will pull back the entire contents of a request, and paginate it client-side. This assumes that your API is returning an array. If it's not returning an array, you can use a transformer (inside the REST API query setup) to make it into an array.


I tried using this, but this restricts the results to the limit we set and it doesn't provide the navigation to move to the next page.
So basically we just get a subset of the results.

**Reply by mjashanks** [reply]
Is the "Paginate" button checked (see screenshot above) ?

**Reply by yatharth-zeotap** [reply]
@mjashanks 
Yes it is checked

**Comment by yatharth-zeotap** [comment]
Hi @melohagan / @mjashanks, I am using an option picker. I want its value to be populated in a text field when I select an option.
Can you please help me with this?
I tried using creating a BUTTON ACTION and using UPDATE STATE to set this value to a state and using this value as a placeholder/default in a text field, but that didn't work.
<img width="1096" alt="image" src="https://user-images.githubusercontent.com/74049974/163093102-c0481774-1ecc-43d3-8974-990e564448e9.png">
<img width="647" alt="image" src="https://user-images.githubusercontent.com/74049974/163093154-1b26ea8b-3cc7-450f-a04b-afc19d004c7a.png">
